From 4690bbea56eff28ac67c0cb9b18418ade070df1c Mon Sep 17 00:00:00 2001
From: Russell Haley <yumpusamongus@gmail.com>
Date: Mon, 20 Mar 2023 20:25:41 -0500
Subject: [PATCH] Fix out-of-bounds index uncovered by -D_GLIBCXX_ASSERTIONS

Flatpak and Fedora packages are built with this flag, and it was causing
a crash on save. Fixes #5570. Also encountered by manolollr in #5387.
---
 src/core/Property.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/Property.cpp b/src/core/Property.cpp
index ccf20480db..2a17b3afef 100644
--- a/src/core/Property.cpp
+++ b/src/core/Property.cpp
@@ -115,7 +115,7 @@ PropertyMap::iterator PropertyMap::iterator::operator++()
 {
 	size_t sz = map->m_keys.size();
 	if (map && sz > idx)
-		while (!map->m_keys[++idx] && idx < sz) {
+		while (++idx < sz && !map->m_keys[idx]) {
 		}
 	return *this;
 }
